react/src/components/SessionUsageMonitor.tsx

```diff
@@ -0,0 +1,260 @@
+import { iSizeToSize, toFixedFloorWithoutTrailingZeros } from '../helper';
+import Flex from './Flex';
+import { ResourceTypeIcon } from './ResourceNumber';
+import { SessionUsageMonitorFragment$key } from './__generated__/SessionUsageMonitorFragment.graphql';
+import { Col, Progress, ProgressProps, Row, Typography, theme } from 'antd';
+import { createStyles } from 'antd-style';
+import graphql from 'babel-plugin-relay/macro';
+import _ from 'lodash';
+import { useMemo } from 'react';
+import { useFragment } from 'react-relay';
+
+const useStyles = createStyles(({ css, token }) => ({
+  progress: css`
+    .ant-progress-text {
+      font-size: ${token.fontSizeSM} !important;
+    }
+    .ant-progress-inner {
+    }
+    .ant-typography {
+      color: ${token.colorTextSecondary} !important;
+      font-size: ${token.fontSizeSM}px !important;
+    }
+  `,
+}));
+
+interface SessionUsageMonitorProps extends ProgressProps {
+  sessionFrgmt: SessionUsageMonitorFragment$key | null;
+  col?: 1 | 2 | 3 | 4;
+  size?: ProgressProps['size'];
+}
+
+// compute session node query returns accelerator type as `[device_name]_[util|mem]`
+// so we need to split via '_' and get the first part to map to the resource type in `resourceTypeIconSrcMap`
+
+type acceleratorType =
+  | 'cuda'
+  | 'rocm'
+  | 'tpu'
+  | 'ipu'
+  | 'atom'
+  | 'atom-plus'
+  | 'gaudi2'
+  | 'warboy'
+  | 'rngd'
+  | 'hyperaccel-lpu';
+
+const resourceTypeMap: {
+  [key in acceleratorType | 'cpu' | 'mem']: string;
+} = {
+  cpu: 'cpu',
+  mem: 'mem',
+  cuda: 'cuda.device',
+  rocm: 'rocm.device',
+  tpu: 'tpu.device',
+  ipu: 'ipu.device',
+  atom: 'atom.device',
+  'atom-plus': 'atom-plus.device',
+  gaudi2: 'gaudi2.device',
+  warboy: 'warboy.device',
+  rngd: 'rngd.device',
+  'hyperaccel-lpu': 'hyperaccel-lpu.device',
+};
```

Comment:
It's better to create a function to add `device` without using hardcoded accelerator names.